index.php

```diff
@@ -187,44 +112,40 @@
                            get_string('backupsize', 'report_coursesize'));
 $coursetable->data = array();
 
-arsort($coursesizes);
 $totalsize = 0;
 $totalbackupsize = 0;
 $downloaddata = array();
 $downloaddata[] = array(get_string('course'),
                            get_string('category'),
                            get_string('diskusage', 'report_coursesize'),
                            get_string('backupsize', 'report_coursesize'));;
-foreach ($coursesizes as $courseid => $size) {
-    if (empty($courses[$courseid])) {
-        continue;
-    }
-    $backupsize = $coursebackupsizes[$courseid];
-    $totalsize = $totalsize + $size;
-    $totalbackupsize  = $totalbackupsize + $backupsize;
-    $course = $courses[$courseid];
+
+$coursesizes = $DB->get_records('report_coursesize');
+foreach ($courses as $courseid => $course) {
+
+    $totalsize = $totalsize + $course->filesize;
+    $totalbackupsize  = $totalbackupsize + $course->backupsize;
     $coursecontext = context_course::instance($course->id);
     $course->shortname = format_string($course->shortname, true, ['context' => $coursecontext]);
     $course->name = format_string($course->name, true, ['context' => $coursecontext]);
     $row = array();
     $row[] = '<a href="'.$CFG->wwwroot.'/course/view.php?id='.$course->id.'">' . $course->shortname . '</a>';
     $row[] = '<a href="'.$CFG->wwwroot.'/course/index.php?categoryid='.$course->category.'">' . $course->name . '</a>';
 
-    $readablesize = number_format(ceil($size / 1000000)) . $sizemb;
+    $readablesize = number_format(ceil($course->filesize / 1000000)) . $sizemb;
```

Comment:
This could be display_size() instead for consistency

Reply:
Heh yeah, I needed to stop somewhere, lots of existing code I'd like to improve further:-)

Reply:
haha yup, easy to go too far

Reply:
this number_format thing is one of the ones that I definitley wanted to fix but it's all the way through the file - I will probably do it, just not in this PR.

Reply:
created #31 to remind me to do this one.